packages/esbuild/src/executors/esbuild/esbuild.impl.ts

```diff
@@ -91,236 +91,103 @@ export async function* esbuildExecutor(
-                            } else {
-                              logger.info(BUILD_WATCH_SUCCEEDED);
-                            }
-
-                            next({
-                              success,
-                              // Need to call getOutfile directly in the case of bundle=false and outfile is not set for esbuild.
-                              outfile: join(
-                                context.root,
-                                getOutfile(format, options, context)
-                              ),
-                            });
-                          });
-                        },
-                      }
-                    : null,
-                ].filter(Boolean),
-              });
-
-              await ctx.watch();
-              return () => ctx.dispose();
-            })
-          );
-
-          registerCleanupCallback(() => {
-            assetsResult?.stop();
-            packageJsonResult?.stop();
-            disposeFns.forEach((fn) => fn());
-            done(); // return from async iterable
-          });
-        }
-      );
-    } else {
-      // Run type-checks first and bail if they don't pass.
-      if (!options.skipTypeCheck) {
-        const { errors } = await runTypeCheck(options, context);
-        if (errors.length > 0) {
-          yield { success: false };
-          return;
-        }
-      }
-
-      // Emit a build event for each file format.
-      for (let i = 0; i < options.format.length; i++) {
-        const format = options.format[i];
-        const esbuildOptions = buildEsbuildOptions(format, options, context);
-        const buildResult = await esbuild.build(esbuildOptions);
-
-        if (options.metafile) {
-          const filename =
-            options.format.length === 1
-              ? 'meta.json'
-              : `meta.${options.format[i]}.json`;
-          writeJsonSync(
-            joinPathFragments(options.outputPath, filename),
-            buildResult.metafile
-          );
-        }
-
-        yield {
-          success: buildResult.errors.length === 0,
-          // Need to call getOutfile directly in the case of bundle=false and outfile is not set for esbuild.
-          // This field is needed for `@nx/js:node` executor to work.
-          outfile: join(context.root, getOutfile(format, options, context)),
-        };
-      }
-    }
-  } else {
-    // TODO(jack): Remove in Nx 16
-    // < 0.17.0 takes watch as an argument to build()
-    if (options.watch) {
-      return yield* createAsyncIterable<{ success: boolean; outfile?: string }>(
-        async ({ next, done }) => {
-          let hasTypeErrors = false;
-          const results = await Promise.all(
-            options.format.map(async (format, idx) => {
-              const esbuildOptions = buildEsbuildOptions(
-                format,
-                options,
-                context
-              );
-              const watch =
+  if (options.watch) {
```

Comment:
This code block is the same as what's in `if ('context' in esbuild) {...}` previously.